services/shhext/api.go

```diff
@@ -474,99 +466,12 @@ func (api *PublicAPI) ConfirmMessagesProcessedByID(messageIDs [][]byte) error {
 
 // SendPublicMessage sends a public chat message to the underlying transport
 func (api *PublicAPI) SendPublicMessage(ctx context.Context, msg chat.SendPublicMessageRPC) (hexutil.Bytes, error) {
-	filter := api.service.filter.GetByID(msg.Chat)
-	if filter == nil {
-		return nil, errors.New("not subscribed to chat")
-	}
-
-	// Enrich with transport layer info
-	whisperMessage := whisperutils.DefaultWhisperMessage()
-	whisperMessage.Payload = msg.Payload
-	whisperMessage.Sig = msg.Sig
-	whisperMessage.Topic = whisperutils.ToTopic(msg.Chat)
-	whisperMessage.SymKeyID = filter.SymKeyID
-
-	// And dispatch
-	return api.Post(ctx, whisperMessage)
-}
-
-func (api *PublicAPI) directMessageToWhisper(myPrivateKey *ecdsa.PrivateKey, theirPublicKey *ecdsa.PublicKey, destination hexutil.Bytes, signature string, spec *chat.ProtocolMessageSpec) (*whisper.NewMessage, error) {
-	// marshal for sending to wire
-	marshaledMessage, err := proto.Marshal(spec.Message)
-	if err != nil {
-		api.log.Error("encryption-service", "error marshaling message", err)
-		return nil, err
-	}
-
-	whisperMessage := whisperutils.DefaultWhisperMessage()
-	whisperMessage.Payload = marshaledMessage
-	whisperMessage.Sig = signature
-
-	if spec.SharedSecret != nil {
-		chat := api.service.GetNegotiatedChat(theirPublicKey)
-		if chat != nil {
-			api.log.Debug("Sending on negotiated topic")
-			whisperMessage.SymKeyID = chat.SymKeyID
-			whisperMessage.Topic = chat.Topic
-			whisperMessage.PublicKey = nil
-			return &whisperMessage, nil
-		}
-	} else if spec.PartitionedTopic() {
-		api.log.Debug("Sending on partitioned topic")
-		// Create filter on demand
-		if _, err := api.service.filter.LoadPartitioned(myPrivateKey, theirPublicKey, false); err != nil {
-			return nil, err
-		}
-		t := filter.PublicKeyToPartitionedTopicBytes(theirPublicKey)
-		whisperMessage.Topic = whisper.BytesToTopic(t)
-		whisperMessage.PublicKey = destination
-		return &whisperMessage, nil
-	}
-
-	api.log.Debug("Sending on old discovery topic")
-	whisperMessage.Topic = whisperutils.DiscoveryTopicBytes
-	whisperMessage.PublicKey = destination
-
-	return &whisperMessage, nil
+	return api.service.SendPublicMessage(ctx, msg)
```

Comment:
so you are planning to plug these method in to status-react already? that's good :-D